### PR TITLE
chore(testing): validate requirements lockfiles with a github action

### PR DESCRIPTION
**Pull request by emmettbutler**
This change adds a github action that fails when a run of `scripts/compile-and-prune-test-requirements` leaves a dirty git tree, indicating that requirements lockfiles need to be generated or updated to match riotfile changes on the branch in question.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] PR description includes explicit acknowledgement/acceptance of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: .github/workflows/requirements-locks.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  validate:
+    name: Check requirements lockfiles
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Ensure riot requirements lockfiles are in sync
+        run: scripts/compile-and-prune-test-requirements && git diff --exit-code
```
Comment (brettlangdon) [inline .github/workflows/requirements-locks.yml:18]: we might need to use our dev docker image for this? we'll need all versions of Python we test with, correct?

Comment (emmettbutler) [reply .github/workflows/requirements-locks.yml:18]: Yup. I'll need to figure out how to do that.

File: .github/workflows/requirements-locks.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install Dependencies
+        run: pip install riot>=0.17.0
+
+      - name: Ensure riot requirements lockfiles are in sync
+        run: scripts/compile-and-prune-test-requirements && pwd && ls && git diff --exit-code
```
Comment (brettlangdon) [inline .github/workflows/requirements-locks.yml:22]: hmm, it didn't seem to like `git diff --exit-code`... but that failed? how is it working in CircleCI?

> riot run -s snapshot-fmt && git diff --exit-code

🤔

**Comment by emmettbutler** [comment]
This is failing due to https://github.com/actions/checkout/issues/363

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**